src/Microsoft.DotNet.Interactive.HttpRequest/Model/HttpResponse.cs

```diff
@@ -1,11 +1,19 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+#nullable enable
+
 using System.Collections.Generic;
+using Microsoft.DotNet.Interactive.Formatting;
 
+#if HTTP_REQUEST_KERNEL
 namespace Microsoft.DotNet.Interactive.HttpRequest;
-
-public class HttpResponse
+[TypeFormatterSource(typeof(HttpResponseFormatterSource))]
```

Comment:
I prefer not to have ifdef. This type is intended also to be used for variable sharing and named request. Maybe it is better discuss visibility.
@jonsequitur ?

Reply:
I think the approach is preferable to the alternatives I can think of, which are:

* Make the `HttpRequest` and `HttpResponse` types public in `Formatting`. (I don't think they belong there, and there's already more stuff there than I'd prefer.)
* Have divergent formatting for e.g. `HttpResponse` and `HttpResponseMessage`. (This is confusing for users and extra maintenance work for us.)

Other ideas?

Reply:
That type is supposed to be the type to use when starting the response to other kernels

Reply:
Side note: Since in the context of the HTTP kernel, this is a message exchange format, we should also add some contract tests for it.